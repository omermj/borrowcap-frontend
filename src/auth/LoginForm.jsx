import { useNavigate } from "react-router-dom";
import { Formik, ErrorMessage } from "formik";
import { Form, Button } from "react-bootstrap";

/** Login Form */

const LoginForm = ({ login }) => {
  const navigate = useNavigate();

  const errMsg = () => (
    <div>
      <small className="text-danger">Incorrect Username/Password</small>
    </div>
  );

  return (
    <div>
      <h3>Login</h3>
      <Formik
        initialValues={{ username: "", password: "" }}
        onSubmit={async (values, { setSubmitting, setErrors }) => {
          const result = await login(values);
          if (result.success) {
            navigate("/");
          } else {
            setErrors({
              username: "Incorrect Username",
              password: "Incorrect Password",
            });
          }
          setSubmitting(false);
          console.log("wrong stuff");
        }}
      >
        {({
          values,
          errors,
          touched,
          handleChange,
          handleBlur,
          handleSubmit,
          isSubmitting,
        }) => (
          <Form onSubmit={handleSubmit}>
            <Form.Group className="mb-3" controlId="loginFormUsername">
              <Form.Label>Username</Form.Label>
              <Form.Control
                type="text"
                name="username"
                placeholder="Enter username"
                value={values.username}
                onChange={handleChange}
                autoComplete="username"
                required
              />
            </Form.Group>
            <Form.Group className="mb-3" controlId="loginFormPassword">
              <Form.Label>Password</Form.Label>
              <Form.Control
                type="password"
                name="password"
                placeholder="Enter password"
                onChange={handleChange}
                value={values.password}
                autoComplete="current-password"
                required
              />
            </Form.Group>
            <ErrorMessage name="username" component={errMsg} className="mt-3" />
            <Button
              variant="primary"
              type="submit"
              disabled={isSubmitting}
              className="mt-4"
            >
              Submit
            </Button>
          </Form>
        )}
      </Formik>
    </div>
  );
};
export default LoginForm;
