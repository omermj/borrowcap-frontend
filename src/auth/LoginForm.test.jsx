import { render } from "@testing-library/react";
import LoginForm from "./LoginForm";
import { BrowserRouter } from "react-router-dom";

// smoke test
it("renders without crashing", () => {
  render(
    <BrowserRouter>
      <LoginForm />
    </BrowserRouter>
  );
});

// snapshot test
it("matches snapshot", () => {
  const { asFragment } = render(
    <BrowserRouter>
      <LoginForm />
    </BrowserRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
