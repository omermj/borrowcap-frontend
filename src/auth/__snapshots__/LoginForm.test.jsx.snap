// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="form-wrapper"
  >
    <div
      class="form-inner"
    >
      <h3
        class="mb-4 text-center"
      >
        Login
      </h3>
      <form
        class=""
      >
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="loginFormUsername"
          >
            Username
          </label>
          <input
            autocomplete="username"
            class="form-control"
            id="loginFormUsername"
            name="username"
            placeholder="Enter username"
            required=""
            type="text"
            value=""
          />
        </div>
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="loginFormPassword"
          >
            Password
          </label>
          <input
            autocomplete="current-password"
            class="form-control"
            id="loginFormPassword"
            name="password"
            placeholder="Enter password"
            required=""
            type="password"
            value=""
          />
        </div>
        <div
          class="d-grid"
        >
          <button
            class="mt-4 btn-dark btn btn-primary"
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
